playground/backend/internal/cloud_bucket/precompiled_objects.go

```diff
@@ -0,0 +1,283 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloud_bucket
+
+import (
+	pb "beam.apache.org/playground/backend/internal/api/v1"
+	"beam.apache.org/playground/backend/internal/logger"
+	"cloud.google.com/go/storage"
+	"context"
+	"encoding/json"
+	"fmt"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const (
+	BucketName       = "test_public_bucket_akvelon"
+	OutputExtension  = "output"
+	MetaInfoName     = "meta.info"
+	Timeout          = 100
+	javaExtension    = "java"
+	goExtension      = "go"
+	pyExtension      = "py"
+	scioExtension    = "scala"
+	separatorsNumber = 2
+)
+
+type ObjectInfo struct {
+	Name        string
+	CloudPath   string
+	Description string                   `protobuf:"bytes,3,opt,name=description,proto3" json:"description,omitempty"`
+	Type        pb.PrecompiledObjectType `protobuf:"varint,4,opt,name=type,proto3,enum=api.v1.PrecompiledObjectType" json:"type,omitempty"`
+	Categories  []string                 `json:"categories,omitempty"`
+}
+
+type PrecompiledObjects []ObjectInfo
+type CategoryToPrecompiledObjects map[string]PrecompiledObjects
+type SdkToCategories map[string]CategoryToPrecompiledObjects
+
+// CloudStorage represents working tools for getting compiled and
+// run beam examples from Google Cloud Storage. It is required that
+// the bucket where examples are stored would be public,
+// and it has a concrete structure of directories, namely:
+// SDK_JAVA/
+// --------PrecompiledObject1/
+// ----------- PrecompiledObject1.java
+// ----------- PrecompiledObject1.output
+// ----------- meta.info
+// --------PrecompiledObject2/
+// ----  ...
+// SDK_GO/
+// --------PrecompiledObject1/
+// ----------- PrecompiledObject1.go
+// ----------- PrecompiledObject1.output
+// ----------- meta.info
+// --------PrecompiledObject2/
+// ----  ...
+// ...
+type CloudStorage struct {
+}
+
+func New() *CloudStorage {
+	return &CloudStorage{}
+}
+
+// GetPrecompiledObject returns the source code of the example
+func (cd *CloudStorage) GetPrecompiledObject(ctx context.Context, precompiledObjectPath string) (*string, error) {
+	extension := getFileExtensionByFileSdk(precompiledObjectPath)
+	data, err := cd.getFileFromBucket(ctx, precompiledObjectPath, extension)
+	if err != nil {
+		return nil, err
+	}
+	result := string(data)
+	return &result, nil
+}
+
+// GetPrecompiledObjectOutput returns the run output of the example
+func (cd *CloudStorage) GetPrecompiledObjectOutput(ctx context.Context, precompiledObjectPath string) (*string, error) {
+	data, err := cd.getFileFromBucket(ctx, precompiledObjectPath, OutputExtension)
+	if err != nil {
+		return nil, err
+	}
+	result := string(data)
+	return &result, nil
+}
+
+// GetPrecompiledObjects returns the list of stored example at cloud storage bucket
```

Comment:
```suggestion
// GetPrecompiledObjects returns the list of stored examples at the cloud storage bucket for the target category
```
I believe, we should add something like this.